Batteries/Data/List/Lemmas.lean

```diff
@@ -1217,6 +1226,17 @@ theorem IsInfix.filter (p : α → Bool) ⦃l₁ l₂ : List α⦄ (h : l₁ <:+
   obtain ⟨xs, ys, rfl⟩ := h
   rw [filter_append, filter_append]; apply infix_append _
 
+theorem cons_prefix_cons : a :: l₁ <+: b :: l₂ ↔ a = b ∧ l₁ <+: l₂ := by
+  constructor
+  · rintro ⟨L, hL⟩
+    simp only [cons_append] at hL
+    injection hL with hLLeft hLRight
+    exact ⟨hLLeft, ⟨L, hLRight⟩⟩
+  · rintro ⟨rfl, h⟩
+    rwa [prefix_cons_inj]
+
+@[deprecated] alias cons_prefix_iff := cons_prefix_cons
```

Comment:
Since `cons_prefix_iff` never existed in Batteries, it doesn't make sense to deprecate here. The deprecation should be in Mathlib.

Reply:
I moved the alias to the corresponding Mathlib PR.